src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -481,38 +477,100 @@ Namespace Microsoft.CodeAnalysis.Operations
             End If
+
+            If operandLazy Is Nothing Then
+                operandLazy = New Lazy(Of IOperation)(Function() CreateConversionOperand(operand))
+            End If
+
+            Return (operandLazy, conversion, IsDelegateCreation(boundConversion.Syntax, operand, boundConversion.Type))
         End Function
 
         ''' <summary>
         ''' Gets the operand and method symbol from a BoundConversion, compensating for if the conversion is a user-defined conversion
         ''' </summary>
-        Private Shared Function GetConversionInfo(boundConversion As BoundConversion) As (Operand As BoundExpression, Method As MethodSymbol)
+        Private Shared Function GetConversionOperand(boundConversion As BoundConversionOrCast) As BoundExpression
             If (boundConversion.ConversionKind And ConversionKind.UserDefined) = ConversionKind.UserDefined Then
                 Dim userDefinedConversion = DirectCast(boundConversion.Operand, BoundUserDefinedConversion)
-                Return (Operand:=userDefinedConversion.Operand, Method:=userDefinedConversion.Call.Method)
+                Return userDefinedConversion.Operand
+            Else
+                Return boundConversion.Operand
+            End If
+        End Function
+
+        Private Function CreateNestedConversionOperand(operand As BoundExpression, conversionSyntax As SyntaxNode) As Lazy(Of IOperation)
+            Return New Lazy(Of IOperation)(Function() CreateNestedConversionOperandCore(operand))
+        End Function
+
+        Private Function CreateNestedConversionOperandCore(operand As BoundExpression) As IOperation
+            Select Case operand.Kind
+                Case BoundKind.Parenthesized
+                    Dim boundParenthesized = DirectCast(operand, BoundParenthesized)
+                    ' Because we're in a delegate creation scenario, the underlying expression actually doesn't have a type.
+                    Dim operandOperation = CreateNestedConversionOperandCore(boundParenthesized.Expression)
+                    Return New ParenthesizedExpression(
+                        operandOperation,
+                        _semanticModel,
+                        boundParenthesized.Syntax,
+                        operandOperation.Type,
+                        ConvertToOptional(boundParenthesized.ConstantValueOpt),
+                        boundParenthesized.WasCompilerGenerated)
+                Case BoundKind.Conversion, BoundKind.DirectCast, BoundKind.TryCast
+                    ' This is the artificial conversion. Erase it from the tree.
+                    Return CreateConversionOperand(DirectCast(operand, BoundConversionOrCast).Operand)
+                Case BoundKind.DelegateCreationExpression
+                    Return CreateBoundDelegateCreationExpressionChildOperation(DirectCast(operand, BoundDelegateCreationExpression))
+                Case Else
+                    Throw ExceptionUtilities.UnexpectedValue(operand.Kind)
```

Comment:
>Throw ExceptionUtilities.UnexpectedValue(operand.Kind) [](start = 20, length = 54)

I am not comfortable about throwing here unless the caller made sure this can never happened. It doesn't look like the caller did.